index.js

```diff
@@ -136,22 +171,37 @@ const build = async (manifest, manifestPath, bundle, runtimeRepo, buildDir, repo
  * @param {string} runtimeRepo The repository used to install the runtime from
  * @param {string} buildDir Where to build the application
  * @param {string} repoName The flatpak repository name
+ * @param {boolean} cacheBuildDir Whether to enable caching the build directory
  */
 const run = async (
     manifestPath,
     runTests,
     bundle,
     runtimeRepo,
     buildDir,
-    repoName
+    repoName,
+    cacheBuildDir,
 ) => {
+    const manifestHash = (await computeHash(manifestPath)).substring(0, 20)
+    const cache_key = `flatpak-builder-${manifestHash}`
```

Comment:
I would guess it's still desirable to hit the cache even when the manifest changes. Say, for example, only one module out of dozens changed, restoring the cache would allow flatpak-builder not to build most of them. Does that sound right to you?

Reply:
That's where the secondary cache keys come to play, the `flatpak-` & `flatpak-builder-` ones; if the action fails to find the specific cache mentioned in the primary key, it will try to restore the secondary ones. I'm not very sure how that works in practice, it definitely needs some testing to ensure we have the desired behaviour

Reply:
I see, thanks for clarifying (I can't resolve this discussion, feel free to do so!)